wdl/src/main/scala/wdl/expression/WdlStandardLibraryFunctions.scala

```diff
@@ -176,6 +176,32 @@ trait WdlStandardLibraryFunctions extends WdlFunctions[WomValue] {
     extractArguments flatMap arrayLength
   }
 
+  def flatten(params: Seq[Try[WomValue]]): Try[WomValue] = {
+    def extractArguments: Try[WomValue] = params.size match {
+      case 1 => params.head
+      case n => Failure(new IllegalArgumentException(s"Invalid number of parameters for engine function flatten(): $n. flatten() takes exactly 1 parameter."))
+    }
+
+    def arrayFlatten(value: WomValue) : Try[WomValue] =
+      try {
+        value match {
+          case WomArray(WomArrayType(WomArrayType(elemType)), arrayValues) =>
+            val strippedDown: Vector[Vector[WomValue]] = arrayValues.map{
+              case WomArray(_, a) => a.toVector
+              case bad => throw new IllegalArgumentException(s"Invalid argument to flatten. Array element ${bad} should have type ${WomArrayType(elemType)}")
```

Comment:
Since you're already verifying the input type, you could probably replace most of this inner logic with a recursive function, eg something like:
```scala
def getFlatValues(v: WomValue): List[WomValue] = v match {
  case WomArray(_, values) => values flatMap getFlatValues
  case other => List(other)
}
```
That should remove the need to have those awkward `throws` inside the method, and has the added benefit of also flattening out deeper-nested array types.